Keybinding test failures on macOS

on macOS 10.13.2
node 8.9.2, npm 5.5.1
yarn 1.3.2
Theia master@812e2ddf111c2086e1282704d592c3938ae94d65

If I do 'yarn test', I get a *lot* of output but some failures seem related to Key bindings

```
@theia/core: Starting server worker...
@theia/core: Server worker has been started. [ID: 3 | PID: 63698]
@theia/core: Server worker is ready to accept messages. [ID: 3 | PID: 63698]
@theia/core: Server worker has been restarted.
@theia/core: Server worker has been disconnected. [ID: 1 | PID: 63596]
@theia/core: Server worker has been stopped. [ID: 1 | PID: 63596]
@theia/core: Server worker has been disconnected. [ID: 3 | PID: 63698]
@theia/core: Server worker has been stopped. [ID: 3 | PID: 63698]
@theia/core:     ✓ start (5986ms)
@theia/core:   file-uri
@theia/core:     ✓ create -> fsPath -> create should be symmetric
@theia/core:     ✓ fsPath -> create -> fsPath should be symmetric
@theia/core:     ✓ from /
@theia/core:     ✓ from //
@theia/core:     ✓ from c:
@theia/core:     ✓ from /c:
@theia/core:     ✓ from /c:/
@theia/core:   69 passing (7s)
@theia/core:   5 failing
@theia/core:   1) keybindings should set a keymap:
@theia/core:       AssertionError: expected false to be true
@theia/core:       + expected - actual
@theia/core:       -false
@theia/core:       +true
@theia/core:       
@theia/core:       at Assertion.<anonymous> (/Users/mark/git/theia/node_modules/chai/lib/chai/core/assertions.js:659:10)
@theia/core:       at Assertion.propertyGetter (/Users/mark/git/theia/node_modules/chai/lib/chai/utils/addProperty.js:62:29)
@theia/core:       at doAsserterAsyncAndAddThen (/Users/mark/git/theia/node_modules/chai-as-promised/lib/chai-as-promised.js:289:22)
@theia/core:       at Assertion.<anonymous> (/Users/mark/git/theia/node_modules/chai-as-promised/lib/chai-as-promised.js:280:43)
@theia/core:       at Assertion.overwritingPropertyGetter (/Users/mark/git/theia/node_modules/chai/lib/chai/utils/overwriteProperty.js:79:37)
@theia/core:       at Object.get (<anonymous>)
@theia/core:       at Object.proxyGetter [as get] (/Users/mark/git/theia/node_modules/chai/lib/chai/utils/proxify.js:86:22)
@theia/core:       at Context.<anonymous> (src/common/keybinding.spec.ts:84:51)
@theia/core:   2) keybindings should reset to default in case of invalid keybinding:
@theia/core:       AssertionError: expected false to be true
@theia/core:       + expected - actual
@theia/core:       -false
@theia/core:       +true
@theia/core:       
@theia/core:       at Assertion.<anonymous> (/Users/mark/git/theia/node_modules/chai/lib/chai/core/assertions.js:659:10)
@theia/core:       at Assertion.propertyGetter (/Users/mark/git/theia/node_modules/chai/lib/chai/utils/addProperty.js:62:29)
@theia/core:       at doAsserterAsyncAndAddThen (/Users/mark/git/theia/node_modules/chai-as-promised/lib/chai-as-promised.js:289:22)
@theia/core:       at Assertion.<anonymous> (/Users/mark/git/theia/node_modules/chai-as-promised/lib/chai-as-promised.js:280:43)
@theia/core:       at Assertion.overwritingPropertyGetter (/Users/mark/git/theia/node_modules/chai/lib/chai/utils/overwriteProperty.js:79:37)
@theia/core:       at Object.get (<anonymous>)
@theia/core:       at Object.proxyGetter [as get] (/Users/mark/git/theia/node_modules/chai/lib/chai/utils/proxify.js:86:22)
@theia/core:       at Context.<anonymous> (src/common/keybinding.spec.ts:100:51)
@theia/core:   3) keybindings should remove all keybindings from a command that has multiple keybindings:
@theia/core:       AssertionError: expected false to be true
@theia/core:       + expected - actual
@theia/core:       -false
@theia/core:       +true
@theia/core:       
@theia/core:       at Assertion.<anonymous> (/Users/mark/git/theia/node_modules/chai/lib/chai/core/assertions.js:659:10)
@theia/core:       at Assertion.propertyGetter (/Users/mark/git/theia/node_modules/chai/lib/chai/utils/addProperty.js:62:29)
@theia/core:       at doAsserterAsyncAndAddThen (/Users/mark/git/theia/node_modules/chai-as-promised/lib/chai-as-promised.js:289:22)
@theia/core:       at Assertion.<anonymous> (/Users/mark/git/theia/node_modules/chai-as-promised/lib/chai-as-promised.js:280:43)
@theia/core:       at Assertion.overwritingPropertyGetter (/Users/mark/git/theia/node_modules/chai/lib/chai/utils/overwriteProperty.js:79:37)
@theia/core:       at Object.get (<anonymous>)
@theia/core:       at Object.proxyGetter [as get] (/Users/mark/git/theia/node_modules/chai/lib/chai/utils/proxify.js:86:22)
@theia/core:       at Context.<anonymous> (src/common/keybinding.spec.ts:116:51)
@theia/core:   4) keybindings should register a correct keybinding, then default back to the original for a wrong one after:
@theia/core:       AssertionError: expected false to be true
@theia/core:       + expected - actual
@theia/core:       -false
@theia/core:       +true
@theia/core:       
@theia/core:       at Assertion.<anonymous> (/Users/mark/git/theia/node_modules/chai/lib/chai/core/assertions.js:659:10)
@theia/core:       at Assertion.propertyGetter (/Users/mark/git/theia/node_modules/chai/lib/chai/utils/addProperty.js:62:29)
@theia/core:       at doAsserterAsyncAndAddThen (/Users/mark/git/theia/node_modules/chai-as-promised/lib/chai-as-promised.js:289:22)
@theia/core:       at Assertion.<anonymous> (/Users/mark/git/theia/node_modules/chai-as-promised/lib/chai-as-promised.js:280:43)
@theia/core:       at Assertion.overwritingPropertyGetter (/Users/mark/git/theia/node_modules/chai/lib/chai/utils/overwriteProperty.js:79:37)
@theia/core:       at Object.get (<anonymous>)
@theia/core:       at Object.proxyGetter [as get] (/Users/mark/git/theia/node_modules/chai/lib/chai/utils/proxify.js:86:22)
@theia/core:       at Context.<anonymous> (src/common/keybinding.spec.ts:133:51)
@theia/core:   5) keys api parses a string to a KeyCode correctly:
@theia/core:       AssertionError: expected false to be true
@theia/core:       + expected - actual
@theia/core:       -false
@theia/core:       +true
@theia/core:       
@theia/core:       at Assertion.<anonymous> (/Users/mark/git/theia/node_modules/chai/lib/chai/core/assertions.js:659:10)
@theia/core:       at Assertion.propertyGetter (/Users/mark/git/theia/node_modules/chai/lib/chai/utils/addProperty.js:62:29)
@theia/core:       at doAsserterAsyncAndAddThen (/Users/mark/git/theia/node_modules/chai-as-promised/lib/chai-as-promised.js:289:22)
@theia/core:       at Assertion.<anonymous> (/Users/mark/git/theia/node_modules/chai-as-promised/lib/chai-as-promised.js:280:43)
@theia/core:       at Assertion.overwritingPropertyGetter (/Users/mark/git/theia/node_modules/chai/lib/chai/utils/overwriteProperty.js:79:37)
@theia/core:       at Object.get (<anonymous>)
@theia/core:       at Object.proxyGetter [as get] (/Users/mark/git/theia/node_modules/chai/lib/chai/utils/proxify.js:86:22)
@theia/core:       at Context.<anonymous> (src/common/keybinding.spec.ts:186:39)
@theia/core: npm ERR! code ELIFECYCLE
@theia/core: npm ERR! errno 5
@theia/core: npm ERR! @theia/core@0.3.0 test: `theiaext test`
@theia/core: npm ERR! Exit status 5
@theia/core: npm ERR! 
@theia/core: npm ERR! Failed at the @theia/core@0.3.0 test script.
@theia/core: npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
@theia/core: npm ERR! A complete log of this run can be found in:
@theia/core: npm ERR!     /Users/mark/.npm/_logs/2017-12-18T20_35_38_733Z-debug.log
lerna ERR! execute Error: Command failed: npm run test
lerna ERR! execute Server worker failed to restart.
lerna ERR! execute npm ERR! code ELIFECYCLE
lerna ERR! execute npm ERR! errno 5
lerna ERR! execute npm ERR! @theia/core@0.3.0 test: `theiaext test`
lerna ERR! execute npm ERR! Exit status 5
lerna ERR! execute npm ERR! 
lerna ERR! execute npm ERR! Failed at the @theia/core@0.3.0 test script.
lerna ERR! execute npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
lerna ERR! execute 
lerna ERR! execute npm ERR! A complete log of this run can be found in:
lerna ERR! execute npm ERR!     /Users/mark/.npm/_logs/2017-12-18T20_35_38_733Z-debug.log
lerna ERR! execute 
lerna ERR! execute 
lerna ERR! execute > @theia/core@0.3.0 test /Users/mark/git/theia/packages/core
lerna ERR! execute > theiaext test
lerna ERR! execute 
lerna ERR! execute 
lerna ERR! execute 
lerna ERR! execute   Endpoint
lerna ERR! execute     01 #getWebSocketUrl
lerna ERR! execute       ✓ Should correctly join root pathname
lerna ERR! execute       ✓ Should correctly join pathname and path
lerna ERR! execute       ✓ Should correctly join pathname and path, ignoring double slash in between
lerna ERR! execute       ✓ Should correctly join pathname and path, without trailing slash
lerna ERR! execute     02 #httpScheme
lerna ERR! execute       ✓ Should choose https:// if location protocol is https://
lerna ERR! execute 
lerna ERR! execute   StatusBarEntryUtility
lerna ERR! execute     ✓ should create an empty array.
lerna ERR! execute     ✓ should create a string array with one entry.
lerna ERR! execute     ✓ should create a string array with one entry - text contains an $.
lerna ERR! execute     ✓ should create a string array with one entry - text contains an $( which does not close.
lerna ERR! execute     ✓ should create a string array with two entries. Second is a simple StatusBarIcon.
lerna ERR! execute     ✓ should create a string array with two entries. Second is a StatusBarIcon with an animation.
lerna ERR! execute     ✓ should create string array of 'foo $(icon1) bar $(icon2) baz $(icon3)'
lerna ERR! execute     ✓ should create string array of '$(icon1) foo bar $(icon2) baz $(icon3)'
lerna ERR! execute     ✓ should create an array with one element of '$(icon1)'
lerna ERR! execute     ✓ should create an array of '$(icon1)$(icon2) (icon3)'
lerna ERR! execute 
lerna ERR! execute   opener-service
lerna ERR! execute     ✓ getOpeners
lerna ERR! execute 
lerna ERR! execute   storage-service
lerna ERR! execute     ✓ stores data
lerna ERR! execute     ✓ removes data
lerna ERR! execute 
lerna ERR! execute   widget-manager
lerna ERR! execute     ✓ creates and caches widgets
lerna ERR! execute 
lerna ERR! execute   keybindings
lerna ERR! execute     ✓ should register the default keybindings
lerna ERR! execute     1) should set a keymap
lerna ERR! execute     2) should reset to default in case of invalid keybinding
lerna ERR! execute     3) should remove all keybindings from a command that has multiple keybindings
lerna ERR! execute     4) should register a correct keybinding, then default back to the original for a wrong one after
lerna ERR! execute     ✓ should only return the more specific keybindings when a keystroke is entered
lerna ERR! execute 
lerna ERR! execute   keys api
lerna ERR! execute     5) parses a string to a KeyCode correctly
lerna ERR! execute 
lerna ERR! execute   menu-model-registry
lerna ERR! execute     01 #register
lerna ERR! execute       ✓ Should allow to register menu actions.
lerna ERR! execute 
lerna ERR! execute   Proxy-Factory
lerna ERR! execute waiting another 50 ms
lerna ERR! execute     ✓ Should correctly send notifications and requests. (73ms)
lerna ERR! execute     ✓ Rejected Promise should result in rejected Promise.
lerna ERR! execute     ✓ Remote Exceptions should result in rejected Promise.
lerna ERR! execute 
lerna ERR! execute   Path
lerna ERR! execute     ✓ new from /foo/bar/file.txt
lerna ERR! execute     ✓ new from foo/bar/file.txt
lerna ERR! execute     ✓ new from /foo
lerna ERR! execute     ✓ new from foo
lerna ERR! execute     ✓ new from /
lerna ERR! execute     ✓ new from /c:/foo/bar/file.txt
lerna ERR! execute     ✓ new from /c:/foo
lerna ERR! execute     ✓ new from /c:/
lerna ERR! execute     ✓ new from /c:
lerna ERR! execute 
lerna ERR! execute   selection-service
lerna ERR! execute     01 #addListener and dispose
lerna ERR! execute       ✓ Should be rejected when path argument is undefined.
lerna ERR! execute 
lerna ERR! execute   types
lerna ERR! execute     Prioritizeable
lerna ERR! execute       ✓ prioritizeAll #01
lerna ERR! execute       ✓ prioritizeAll #02
lerna ERR! execute 
lerna ERR! execute   uri
lerna ERR! execute     #getParent
lerna ERR! execute       ✓ of file:///foo/bar.txt
lerna ERR! execute       ✓ of file:///foo/
lerna ERR! execute       ✓ of file:///foo
lerna ERR! execute       ✓ of file:///
lerna ERR! execute       ✓ of file://
lerna ERR! execute     #lastSegment
lerna ERR! execute       ✓ of file:///foo/bar.txt
lerna ERR! execute       ✓ of file:///foo
lerna ERR! execute       ✓ of file:///
lerna ERR! execute       ✓ of file://
lerna ERR! execute     #appendPath
lerna ERR! execute       ✓ '' to file:///foo
lerna ERR! execute       ✓ bar to file:///foo
lerna ERR! execute       ✓ bar/baz to file:///foo
lerna ERR! execute       ✓ '' to file:///
lerna ERR! execute       ✓ bar to file:///
lerna ERR! execute       ✓ bar/baz to file:///
lerna ERR! execute     #path
lerna ERR! execute       ✓ Should return with the FS path from the URI.
lerna ERR! execute       ✓ Should not return the encoded path
lerna ERR! execute     #withFragment
lerna ERR! execute       ✓ Should replace the fragment.
lerna ERR! execute       ✓ Should remove the fragment.
lerna ERR! execute     #toString()
lerna ERR! execute       ✓ should produce the non encoded string
lerna ERR! execute     #Uri.with...()
lerna ERR! execute       ✓ produce proper URIs
lerna ERR! execute 
lerna ERR! execute   CliManager
lerna ERR! execute     ✓ Parses simple option
lerna ERR! execute     ✓ resolves with default
lerna ERR! execute Options:
lerna ERR! execute   --foo, -f, -f  Some foo.
lerna ERR! execute   --bar, -b, -b  Some bla.                      [string] [default: "my-default"]
lerna ERR! execute   --version      Show version number                                   [boolean]
lerna ERR! execute   --help         Show help                                             [boolean]
lerna ERR! execute 
lerna ERR! execute     ✓ prints help and exits
lerna ERR! execute 
lerna ERR! execute   master-process
lerna ERR! execute Starting server worker...
lerna ERR! execute Server worker has been started. [ID: 1 | PID: 63596]
lerna ERR! execute Server worker is ready to accept messages. [ID: 1 | PID: 63596]
lerna ERR! execute Restarting the server worker is requested.
lerna ERR! execute Starting server worker...
lerna ERR! execute Server worker has been started. [ID: 2 | PID: 63604]
lerna ERR! execute Server worker has been disconnected. [ID: 2 | PID: 63604]
lerna ERR! execute Server worker has been stopped. [ID: 2 | PID: 63604]
lerna ERR! execute Restarting the server worker is requested.
lerna ERR! execute Starting server worker...
lerna ERR! execute Server worker has been started. [ID: 3 | PID: 63698]
lerna ERR! execute Server worker is ready to accept messages. [ID: 3 | PID: 63698]
lerna ERR! execute Server worker has been restarted.
lerna ERR! execute Server worker has been disconnected. [ID: 1 | PID: 63596]
lerna ERR! execute Server worker has been stopped. [ID: 1 | PID: 63596]
lerna ERR! execute Server worker has been disconnected. [ID: 3 | PID: 63698]
lerna ERR! execute Server worker has been stopped. [ID: 3 | PID: 63698]
lerna ERR! execute     ✓ start (5986ms)
lerna ERR! execute 
lerna ERR! execute   file-uri
lerna ERR! execute     ✓ create -> fsPath -> create should be symmetric
lerna ERR! execute     ✓ fsPath -> create -> fsPath should be symmetric
lerna ERR! execute     ✓ from /
lerna ERR! execute     ✓ from //
lerna ERR! execute     ✓ from c:
lerna ERR! execute     ✓ from /c:
lerna ERR! execute     ✓ from /c:/
lerna ERR! execute 
lerna ERR! execute 
lerna ERR! execute   69 passing (7s)
lerna ERR! execute   5 failing
lerna ERR! execute 
lerna ERR! execute   1) keybindings should set a keymap:
lerna ERR! execute 
lerna ERR! execute       AssertionError: expected false to be true
lerna ERR! execute       + expected - actual
lerna ERR! execute 
lerna ERR! execute       -false
lerna ERR! execute       +true
lerna ERR! execute       
lerna ERR! execute       at Assertion.<anonymous> (/Users/mark/git/theia/node_modules/chai/lib/chai/core/assertions.js:659:10)
lerna ERR! execute       at Assertion.propertyGetter (/Users/mark/git/theia/node_modules/chai/lib/chai/utils/addProperty.js:62:29)
lerna ERR! execute       at doAsserterAsyncAndAddThen (/Users/mark/git/theia/node_modules/chai-as-promised/lib/chai-as-promised.js:289:22)
lerna ERR! execute       at Assertion.<anonymous> (/Users/mark/git/theia/node_modules/chai-as-promised/lib/chai-as-promised.js:280:43)
lerna ERR! execute       at Assertion.overwritingPropertyGetter (/Users/mark/git/theia/node_modules/chai/lib/chai/utils/overwriteProperty.js:79:37)
lerna ERR! execute       at Object.get (<anonymous>)
lerna ERR! execute       at Object.proxyGetter [as get] (/Users/mark/git/theia/node_modules/chai/lib/chai/utils/proxify.js:86:22)
lerna ERR! execute       at Context.<anonymous> (src/common/keybinding.spec.ts:84:51)
lerna ERR! execute 
lerna ERR! execute   2) keybindings should reset to default in case of invalid keybinding:
lerna ERR! execute 
lerna ERR! execute       AssertionError: expected false to be true
lerna ERR! execute       + expected - actual
lerna ERR! execute 
lerna ERR! execute       -false
lerna ERR! execute       +true
lerna ERR! execute       
lerna ERR! execute       at Assertion.<anonymous> (/Users/mark/git/theia/node_modules/chai/lib/chai/core/assertions.js:659:10)
lerna ERR! execute       at Assertion.propertyGetter (/Users/mark/git/theia/node_modules/chai/lib/chai/utils/addProperty.js:62:29)
lerna ERR! execute       at doAsserterAsyncAndAddThen (/Users/mark/git/theia/node_modules/chai-as-promised/lib/chai-as-promised.js:289:22)
lerna ERR! execute       at Assertion.<anonymous> (/Users/mark/git/theia/node_modules/chai-as-promised/lib/chai-as-promised.js:280:43)
lerna ERR! execute       at Assertion.overwritingPropertyGetter (/Users/mark/git/theia/node_modules/chai/lib/chai/utils/overwriteProperty.js:79:37)
lerna ERR! execute       at Object.get (<anonymous>)
lerna ERR! execute       at Object.proxyGetter [as get] (/Users/mark/git/theia/node_modules/chai/lib/chai/utils/proxify.js:86:22)
lerna ERR! execute       at Context.<anonymous> (src/common/keybinding.spec.ts:100:51)
lerna ERR! execute 
lerna ERR! execute   3) keybindings should remove all keybindings from a command that has multiple keybindings:
lerna ERR! execute 
lerna ERR! execute       AssertionError: expected false to be true
lerna ERR! execute       + expected - actual
lerna ERR! execute 
lerna ERR! execute       -false
lerna ERR! execute       +true
lerna ERR! execute       
lerna ERR! execute       at Assertion.<anonymous> (/Users/mark/git/theia/node_modules/chai/lib/chai/core/assertions.js:659:10)
lerna ERR! execute       at Assertion.propertyGetter (/Users/mark/git/theia/node_modules/chai/lib/chai/utils/addProperty.js:62:29)
lerna ERR! execute       at doAsserterAsyncAndAddThen (/Users/mark/git/theia/node_modules/chai-as-promised/lib/chai-as-promised.js:289:22)
lerna ERR! execute       at Assertion.<anonymous> (/Users/mark/git/theia/node_modules/chai-as-promised/lib/chai-as-promised.js:280:43)
lerna ERR! execute       at Assertion.overwritingPropertyGetter (/Users/mark/git/theia/node_modules/chai/lib/chai/utils/overwriteProperty.js:79:37)
lerna ERR! execute       at Object.get (<anonymous>)
lerna ERR! execute       at Object.proxyGetter [as get] (/Users/mark/git/theia/node_modules/chai/lib/chai/utils/proxify.js:86:22)
lerna ERR! execute       at Context.<anonymous> (src/common/keybinding.spec.ts:116:51)
lerna ERR! execute 
lerna ERR! execute   4) keybindings should register a correct keybinding, then default back to the original for a wrong one after:
lerna ERR! execute 
lerna ERR! execute       AssertionError: expected false to be true
lerna ERR! execute       + expected - actual
lerna ERR! execute 
lerna ERR! execute       -false
lerna ERR! execute       +true
lerna ERR! execute       
lerna ERR! execute       at Assertion.<anonymous> (/Users/mark/git/theia/node_modules/chai/lib/chai/core/assertions.js:659:10)
lerna ERR! execute       at Assertion.propertyGetter (/Users/mark/git/theia/node_modules/chai/lib/chai/utils/addProperty.js:62:29)
lerna ERR! execute       at doAsserterAsyncAndAddThen (/Users/mark/git/theia/node_modules/chai-as-promised/lib/chai-as-promised.js:289:22)
lerna ERR! execute       at Assertion.<anonymous> (/Users/mark/git/theia/node_modules/chai-as-promised/lib/chai-as-promised.js:280:43)
lerna ERR! execute       at Assertion.overwritingPropertyGetter (/Users/mark/git/theia/node_modules/chai/lib/chai/utils/overwriteProperty.js:79:37)
lerna ERR! execute       at Object.get (<anonymous>)
lerna ERR! execute       at Object.proxyGetter [as get] (/Users/mark/git/theia/node_modules/chai/lib/chai/utils/proxify.js:86:22)
lerna ERR! execute       at Context.<anonymous> (src/common/keybinding.spec.ts:133:51)
lerna ERR! execute 
lerna ERR! execute   5) keys api parses a string to a KeyCode correctly:
lerna ERR! execute 
lerna ERR! execute       AssertionError: expected false to be true
lerna ERR! execute       + expected - actual
lerna ERR! execute 
lerna ERR! execute       -false
lerna ERR! execute       +true
lerna ERR! execute       
lerna ERR! execute       at Assertion.<anonymous> (/Users/mark/git/theia/node_modules/chai/lib/chai/core/assertions.js:659:10)
lerna ERR! execute       at Assertion.propertyGetter (/Users/mark/git/theia/node_modules/chai/lib/chai/utils/addProperty.js:62:29)
lerna ERR! execute       at doAsserterAsyncAndAddThen (/Users/mark/git/theia/node_modules/chai-as-promised/lib/chai-as-promised.js:289:22)
lerna ERR! execute       at Assertion.<anonymous> (/Users/mark/git/theia/node_modules/chai-as-promised/lib/chai-as-promised.js:280:43)
lerna ERR! execute       at Assertion.overwritingPropertyGetter (/Users/mark/git/theia/node_modules/chai/lib/chai/utils/overwriteProperty.js:79:37)
lerna ERR! execute       at Object.get (<anonymous>)
lerna ERR! execute       at Object.proxyGetter [as get] (/Users/mark/git/theia/node_modules/chai/lib/chai/utils/proxify.js:86:22)
lerna ERR! execute       at Context.<anonymous> (src/common/keybinding.spec.ts:186:39)
lerna ERR! execute 
lerna ERR! execute 
lerna ERR! execute 
lerna ERR! execute 
lerna ERR! execute     at Promise.all.then.arr (/Users/mark/git/theia/node_modules/execa/index.js:236:11)
lerna ERR! execute     at <anonymous>
lerna ERR! execute     at process._tickCallback (internal/process/next_tick.js:188:7)
lerna ERR! execute  callback with error
lerna ERR! execute  { Error: Command failed: npm run test
lerna ERR! execute Server worker failed to restart.
lerna ERR! execute npm ERR! code ELIFECYCLE
lerna ERR! execute npm ERR! errno 5
lerna ERR! execute npm ERR! @theia/core@0.3.0 test: `theiaext test`
lerna ERR! execute npm ERR! Exit status 5
lerna ERR! execute npm ERR! 
lerna ERR! execute npm ERR! Failed at the @theia/core@0.3.0 test script.
lerna ERR! execute npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
lerna ERR! execute 
lerna ERR! execute npm ERR! A complete log of this run can be found in:
lerna ERR! execute npm ERR!     /Users/mark/.npm/_logs/2017-12-18T20_35_38_733Z-debug.log
lerna ERR! execute 
lerna ERR! execute 
lerna ERR! execute > @theia/core@0.3.0 test /Users/mark/git/theia/packages/core
lerna ERR! execute > theiaext test
lerna ERR! execute 
lerna ERR! execute 
lerna ERR! execute 
lerna ERR! execute   Endpoint
lerna ERR! execute     01 #getWebSocketUrl
lerna ERR! execute       ✓ Should correctly join root pathname
lerna ERR! execute       ✓ Should correctly join pathname and path
lerna ERR! execute       ✓ Should correctly join pathname and path, ignoring double slash in between
lerna ERR! execute       ✓ Should correctly join pathname and path, without trailing slash
lerna ERR! execute     02 #httpScheme
lerna ERR! execute       ✓ Should choose https:// if location protocol is https://
lerna ERR! execute 
lerna ERR! execute   StatusBarEntryUtility
lerna ERR! execute     ✓ should create an empty array.
lerna ERR! execute     ✓ should create a string array with one entry.
lerna ERR! execute     ✓ should create a string array with one entry - text contains an $.
lerna ERR! execute     ✓ should create a string array with one entry - text contains an $( which does not close.
lerna ERR! execute     ✓ should create a string array with two entries. Second is a simple StatusBarIcon.
lerna ERR! execute     ✓ should create a string array with two entries. Second is a StatusBarIcon with an animation.
lerna ERR! execute     ✓ should create string array of 'foo $(icon1) bar $(icon2) baz $(icon3)'
lerna ERR! execute     ✓ should create string array of '$(icon1) foo bar $(icon2) baz $(icon3)'
lerna ERR! execute     ✓ should create an array with one element of '$(icon1)'
lerna ERR! execute     ✓ should create an array of '$(icon1)$(icon2) (icon3)'
lerna ERR! execute 
lerna ERR! execute   opener-service
lerna ERR! execute     ✓ getOpeners
lerna ERR! execute 
lerna ERR! execute   storage-service
lerna ERR! execute     ✓ stores data
lerna ERR! execute     ✓ removes data
lerna ERR! execute 
lerna ERR! execute   widget-manager
lerna ERR! execute     ✓ creates and caches widgets
lerna ERR! execute 
lerna ERR! execute   keybindings
lerna ERR! execute     ✓ should register the default keybindings
lerna ERR! execute     1) should set a keymap
lerna ERR! execute     2) should reset to default in case of invalid keybinding
lerna ERR! execute     3) should remove all keybindings from a command that has multiple keybindings
lerna ERR! execute     4) should register a correct keybinding, then default back to the original for a wrong one after
lerna ERR! execute     ✓ should only return the more specific keybindings when a keystroke is entered
lerna ERR! execute 
lerna ERR! execute   keys api
lerna ERR! execute     5) parses a string to a KeyCode correctly
lerna ERR! execute 
lerna ERR! execute   menu-model-registry
lerna ERR! execute     01 #register
lerna ERR! execute       ✓ Should allow to register menu actions.
lerna ERR! execute 
lerna ERR! execute   Proxy-Factory
lerna ERR! execute waiting another 50 ms
lerna ERR! execute     ✓ Should correctly send notifications and requests. (73ms)
lerna ERR! execute     ✓ Rejected Promise should result in rejected Promise.
lerna ERR! execute     ✓ Remote Exceptions should result in rejected Promise.
lerna ERR! execute 
lerna ERR! execute   Path
lerna ERR! execute     ✓ new from /foo/bar/file.txt
lerna ERR! execute     ✓ new from foo/bar/file.txt
lerna ERR! execute     ✓ new from /foo
lerna ERR! execute     ✓ new from foo
lerna ERR! execute     ✓ new from /
lerna ERR! execute     ✓ new from /c:/foo/bar/file.txt
lerna ERR! execute     ✓ new from /c:/foo
lerna ERR! execute     ✓ new from /c:/
lerna ERR! execute     ✓ new from /c:
lerna ERR! execute 
lerna ERR! execute   selection-service
lerna ERR! execute     01 #addListener and dispose
lerna ERR! execute       ✓ Should be rejected when path argument is undefined.
lerna ERR! execute 
lerna ERR! execute   types
lerna ERR! execute     Prioritizeable
lerna ERR! execute       ✓ prioritizeAll #01
lerna ERR! execute       ✓ prioritizeAll #02
lerna ERR! execute 
lerna ERR! execute   uri
lerna ERR! execute     #getParent
lerna ERR! execute       ✓ of file:///foo/bar.txt
lerna ERR! execute       ✓ of file:///foo/
lerna ERR! execute       ✓ of file:///foo
lerna ERR! execute       ✓ of file:///
lerna ERR! execute       ✓ of file://
lerna ERR! execute     #lastSegment
lerna ERR! execute       ✓ of file:///foo/bar.txt
lerna ERR! execute       ✓ of file:///foo
lerna ERR! execute       ✓ of file:///
lerna ERR! execute       ✓ of file://
lerna ERR! execute     #appendPath
lerna ERR! execute       ✓ '' to file:///foo
lerna ERR! execute       ✓ bar to file:///foo
lerna ERR! execute       ✓ bar/baz to file:///foo
lerna ERR! execute       ✓ '' to file:///
lerna ERR! execute       ✓ bar to file:///
lerna ERR! execute       ✓ bar/baz to file:///
lerna ERR! execute     #path
lerna ERR! execute       ✓ Should return with the FS path from the URI.
lerna ERR! execute       ✓ Should not return the encoded path
lerna ERR! execute     #withFragment
lerna ERR! execute       ✓ Should replace the fragment.
lerna ERR! execute       ✓ Should remove the fragment.
lerna ERR! execute     #toString()
lerna ERR! execute       ✓ should produce the non encoded string
lerna ERR! execute     #Uri.with...()
lerna ERR! execute       ✓ produce proper URIs
lerna ERR! execute 
lerna ERR! execute   CliManager
lerna ERR! execute     ✓ Parses simple option
lerna ERR! execute     ✓ resolves with default
lerna ERR! execute Options:
lerna ERR! execute   --foo, -f, -f  Some foo.
lerna ERR! execute   --bar, -b, -b  Some bla.                      [string] [default: "my-default"]
lerna ERR! execute   --version      Show version number                                   [boolean]
lerna ERR! execute   --help         Show help                                             [boolean]
lerna ERR! execute 
lerna ERR! execute     ✓ prints help and exits
lerna ERR! execute 
lerna ERR! execute   master-process
lerna ERR! execute Starting server worker...
lerna ERR! execute Server worker has been started. [ID: 1 | PID: 63596]
lerna ERR! execute Server worker is ready to accept messages. [ID: 1 | PID: 63596]
lerna ERR! execute Restarting the server worker is requested.
lerna ERR! execute Starting server worker...
lerna ERR! execute Server worker has been started. [ID: 2 | PID: 63604]
lerna ERR! execute Server worker has been disconnected. [ID: 2 | PID: 63604]
lerna ERR! execute Server worker has been stopped. [ID: 2 | PID: 63604]
lerna ERR! execute Restarting the server worker is requested.
lerna ERR! execute Starting server worker...
lerna ERR! execute Server worker has been started. [ID: 3 | PID: 63698]
lerna ERR! execute Server worker is ready to accept messages. [ID: 3 | PID: 63698]
lerna ERR! execute Server worker has been restarted.
lerna ERR! execute Server worker has been disconnected. [ID: 1 | PID: 63596]
lerna ERR! execute Server worker has been stopped. [ID: 1 | PID: 63596]
lerna ERR! execute Server worker has been disconnected. [ID: 3 | PID: 63698]
lerna ERR! execute Server worker has been stopped. [ID: 3 | PID: 63698]
lerna ERR! execute     ✓ start (5986ms)
lerna ERR! execute 
lerna ERR! execute   file-uri
lerna ERR! execute     ✓ create -> fsPath -> create should be symmetric
lerna ERR! execute     ✓ fsPath -> create -> fsPath should be symmetric
lerna ERR! execute     ✓ from /
lerna ERR! execute     ✓ from //
lerna ERR! execute     ✓ from c:
lerna ERR! execute     ✓ from /c:
lerna ERR! execute     ✓ from /c:/
lerna ERR! execute 
lerna ERR! execute 
lerna ERR! execute   69 passing (7s)
lerna ERR! execute   5 failing
lerna ERR! execute 
lerna ERR! execute   1) keybindings should set a keymap:
lerna ERR! execute 
lerna ERR! execute       AssertionError: expected false to be true
lerna ERR! execute       + expected - actual
lerna ERR! execute 
lerna ERR! execute       -false
lerna ERR! execute       +true
lerna ERR! execute       
lerna ERR! execute       at Assertion.<anonymous> (/Users/mark/git/theia/node_modules/chai/lib/chai/core/assertions.js:659:10)
lerna ERR! execute       at Assertion.propertyGetter (/Users/mark/git/theia/node_modules/chai/lib/chai/utils/addProperty.js:62:29)
lerna ERR! execute       at doAsserterAsyncAndAddThen (/Users/mark/git/theia/node_modules/chai-as-promised/lib/chai-as-promised.js:289:22)
lerna ERR! execute       at Assertion.<anonymous> (/Users/mark/git/theia/node_modules/chai-as-promised/lib/chai-as-promised.js:280:43)
lerna ERR! execute       at Assertion.overwritingPropertyGetter (/Users/mark/git/theia/node_modules/chai/lib/chai/utils/overwriteProperty.js:79:37)
lerna ERR! execute       at Object.get (<anonymous>)
lerna ERR! execute       at Object.proxyGetter [as get] (/Users/mark/git/theia/node_modules/chai/lib/chai/utils/proxify.js:86:22)
lerna ERR! execute       at Context.<anonymous> (src/common/keybinding.spec.ts:84:51)
lerna ERR! execute 
lerna ERR! execute   2) keybindings should reset to default in case of invalid keybinding:
lerna ERR! execute 
lerna ERR! execute       AssertionError: expected false to be true
lerna ERR! execute       + expected - actual
lerna ERR! execute 
lerna ERR! execute       -false
lerna ERR! execute       +true
lerna ERR! execute       
lerna ERR! execute       at Assertion.<anonymous> (/Users/mark/git/theia/node_modules/chai/lib/chai/core/assertions.js:659:10)
lerna ERR! execute       at Assertion.propertyGetter (/Users/mark/git/theia/node_modules/chai/lib/chai/utils/addProperty.js:62:29)
lerna ERR! execute       at doAsserterAsyncAndAddThen (/Users/mark/git/theia/node_modules/chai-as-promised/lib/chai-as-promised.js:289:22)
lerna ERR! execute       at Assertion.<anonymous> (/Users/mark/git/theia/node_modules/chai-as-promised/lib/chai-as-promised.js:280:43)
lerna ERR! execute       at Assertion.overwritingPropertyGetter (/Users/mark/git/theia/node_modules/chai/lib/chai/utils/overwriteProperty.js:79:37)
lerna ERR! execute       at Object.get (<anonymous>)
lerna ERR! execute       at Object.proxyGetter [as get] (/Users/mark/git/theia/node_modules/chai/lib/chai/utils/proxify.js:86:22)
lerna ERR! execute       at Context.<anonymous> (src/common/keybinding.spec.ts:100:51)
lerna ERR! execute 
lerna ERR! execute   3) keybindings should remove all keybindings from a command that has multiple keybindings:
lerna ERR! execute 
lerna ERR! execute       AssertionError: expected false to be true
lerna ERR! execute       + expected - actual
lerna ERR! execute 
lerna ERR! execute       -false
lerna ERR! execute       +true
lerna ERR! execute       
lerna ERR! execute       at Assertion.<anonymous> (/Users/mark/git/theia/node_modules/chai/lib/chai/core/assertions.js:659:10)
lerna ERR! execute       at Assertion.propertyGetter (/Users/mark/git/theia/node_modules/chai/lib/chai/utils/addProperty.js:62:29)
lerna ERR! execute       at doAsserterAsyncAndAddThen (/Users/mark/git/theia/node_modules/chai-as-promised/lib/chai-as-promised.js:289:22)
lerna ERR! execute       at Assertion.<anonymous> (/Users/mark/git/theia/node_modules/chai-as-promised/lib/chai-as-promised.js:280:43)
lerna ERR! execute       at Assertion.overwritingPropertyGetter (/Users/mark/git/theia/node_modules/chai/lib/chai/utils/overwriteProperty.js:79:37)
lerna ERR! execute       at Object.get (<anonymous>)
lerna ERR! execute       at Object.proxyGetter [as get] (/Users/mark/git/theia/node_modules/chai/lib/chai/utils/proxify.js:86:22)
lerna ERR! execute       at Context.<anonymous> (src/common/keybinding.spec.ts:116:51)
lerna ERR! execute 
lerna ERR! execute   4) keybindings should register a correct keybinding, then default back to the original for a wrong one after:
lerna ERR! execute 
lerna ERR! execute       AssertionError: expected false to be true
lerna ERR! execute       + expected - actual
lerna ERR! execute 
lerna ERR! execute       -false
lerna ERR! execute       +true
lerna ERR! execute       
lerna ERR! execute       at Assertion.<anonymous> (/Users/mark/git/theia/node_modules/chai/lib/chai/core/assertions.js:659:10)
lerna ERR! execute       at Assertion.propertyGetter (/Users/mark/git/theia/node_modules/chai/lib/chai/utils/addProperty.js:62:29)
lerna ERR! execute       at doAsserterAsyncAndAddThen (/Users/mark/git/theia/node_modules/chai-as-promised/lib/chai-as-promised.js:289:22)
lerna ERR! execute       at Assertion.<anonymous> (/Users/mark/git/theia/node_modules/chai-as-promised/lib/chai-as-promised.js:280:43)
lerna ERR! execute       at Assertion.overwritingPropertyGetter (/Users/mark/git/theia/node_modules/chai/lib/chai/utils/overwriteProperty.js:79:37)
lerna ERR! execute       at Object.get (<anonymous>)
lerna ERR! execute       at Object.proxyGetter [as get] (/Users/mark/git/theia/node_modules/chai/lib/chai/utils/proxify.js:86:22)
lerna ERR! execute       at Context.<anonymous> (src/common/keybinding.spec.ts:133:51)
lerna ERR! execute 
lerna ERR! execute   5) keys api parses a string to a KeyCode correctly:
lerna ERR! execute 
lerna ERR! execute       AssertionError: expected false to be true
lerna ERR! execute       + expected - actual
lerna ERR! execute 
lerna ERR! execute       -false
lerna ERR! execute       +true
lerna ERR! execute       
lerna ERR! execute       at Assertion.<anonymous> (/Users/mark/git/theia/node_modules/chai/lib/chai/core/assertions.js:659:10)
lerna ERR! execute       at Assertion.propertyGetter (/Users/mark/git/theia/node_modules/chai/lib/chai/utils/addProperty.js:62:29)
lerna ERR! execute       at doAsserterAsyncAndAddThen (/Users/mark/git/theia/node_modules/chai-as-promised/lib/chai-as-promised.js:289:22)
lerna ERR! execute       at Assertion.<anonymous> (/Users/mark/git/theia/node_modules/chai-as-promised/lib/chai-as-promised.js:280:43)
lerna ERR! execute       at Assertion.overwritingPropertyGetter (/Users/mark/git/theia/node_modules/chai/lib/chai/utils/overwriteProperty.js:79:37)
lerna ERR! execute       at Object.get (<anonymous>)
lerna ERR! execute       at Object.proxyGetter [as get] (/Users/mark/git/theia/node_modules/chai/lib/chai/utils/proxify.js:86:22)
lerna ERR! execute       at Context.<anonymous> (src/common/keybinding.spec.ts:186:39)
lerna ERR! execute 
lerna ERR! execute 
lerna ERR! execute 
lerna ERR! execute 
lerna ERR! execute     at Promise.all.then.arr (/Users/mark/git/theia/node_modules/execa/index.js:236:11)
lerna ERR! execute     at <anonymous>
lerna ERR! execute     at process._tickCallback (internal/process/next_tick.js:188:7)
```

--- a/packages/core/src/common/keybinding.spec.ts
+++ b/packages/core/src/common/keybinding.spec.ts
@@ -1,3 +1,4 @@
+/* tslint:disable:no-unused-expression */
 /*
  * Copyright (C) 2017 Ericsson and others.
  *
@@ -12,6 +13,7 @@ import { KeybindingRegistry, KeybindingContext, KeybindingContextRegistry, Keybi
 import { KeyCode, Key, Modifier } from './keys';
 import { CommandRegistry, CommandContribution, Command } from './command';
 import { MockLogger } from './test/mock-logger';
+import { isOSX } from './os';
 import * as chai from 'chai';
 const jsdom = require('jsdom-global');
 
@@ -73,7 +75,7 @@ describe('keybindings', () => {
     it("should set a keymap", () => {
         const rawKeybindings: RawKeybinding[] = [{
             command: "test.command",
-            keybinding: "ctrl+c"
+            keybinding: "shift+c"
         }];
 
         keybindingRegistry.setKeymap(KeybindingScope.USER, rawKeybindings);
@@ -81,7 +83,7 @@ describe('keybindings', () => {
         const bindings = keybindingRegistry.getKeybindingsForCommand('test.command');
         if (bindings) {
             expect(bindings[0].keyCode.key).to.be.equal(Key.KEY_C);
-            expect(bindings[0].keyCode.ctrl).to.be.true;
+            expect(bindings[0].keyCode.shift).to.be.true;
         }
 
     });
@@ -89,7 +91,7 @@ describe('keybindings', () => {
     it("should reset to default in case of invalid keybinding", () => {
         const rawKeybindings: RawKeybinding[] = [{
             command: "test.command",
-            keybinding: "ctrl+invalid"
+            keybinding: "shift+invalid"
         }];
 
         keybindingRegistry.setKeymap(KeybindingScope.USER, rawKeybindings);
@@ -97,7 +99,7 @@ describe('keybindings', () => {
         const bindings = keybindingRegistry.getKeybindingsForCommand('test.command');
         if (bindings) {
             expect(bindings[0].keyCode.key).to.be.equal(Key.KEY_A);
-            expect(bindings[0].keyCode.ctrl).to.be.true;
+            expect(bindings[0].keyCode.shift).to.be.true;
         }
     });
 
@@ -113,14 +115,14 @@ describe('keybindings', () => {
         if (bindings) {
             expect(bindings.length).to.be.equal(2);
             expect(bindings[0].keyCode.key).to.be.equal(Key.F1);
-            expect(bindings[0].keyCode.ctrl).to.be.true;
+            expect(bindings[0].keyCode.shift).to.be.true;
         }
     });
 
     it("should register a correct keybinding, then default back to the original for a wrong one after", () => {
         let rawKeybindings: RawKeybinding[] = [{
             command: "test.command",
-            keybinding: "ctrl+c"
+            keybinding: "shift+c"
         }];
         // Get default binding
         const keystroke = keybindingRegistry.getKeybindingsForCommand('test.command');
@@ -130,7 +132,7 @@ describe('keybindings', () => {
         const bindings = keybindingRegistry.getKeybindingsForCommand('test.command');
         if (bindings) {
             expect(bindings[0].keyCode.key).to.be.equal(Key.KEY_C);
-            expect(bindings[0].keyCode.ctrl).to.be.true;
+            expect(bindings[0].keyCode.shift).to.be.true;
         }
 
         // Set invalid binding
@@ -152,43 +154,77 @@ describe('keybindings', () => {
     it("should only return the more specific keybindings when a keystroke is entered", () => {
         const rawKeybindingsUser: RawKeybinding[] = [{
             command: "test.command",
-            keybinding: "ctrl+b"
+            keybinding: "shift+b"
         }];
 
         keybindingRegistry.setKeymap(KeybindingScope.USER, rawKeybindingsUser);
 
         const rawKeybindingsSpecific: RawKeybinding[] = [{
             command: "test.command",
-            keybinding: "ctrl+c"
+            keybinding: "shift+c"
         }];
 
-        const validKeyCode = KeyCode.createKeyCode({ first: Key.KEY_C, modifiers: [Modifier.M1] });
+        const validKeyCode = KeyCode.createKeyCode({ first: Key.KEY_C, modifiers: [Modifier.M2] });
 
         keybindingRegistry.setKeymap(KeybindingScope.WORKSPACE, rawKeybindingsSpecific);
 
-        let bindings = keybindingRegistry.getKeybindingsForKeyCode(KeyCode.createKeyCode({ first: Key.KEY_A, modifiers: [Modifier.M1] }));
+        let bindings = keybindingRegistry.getKeybindingsForKeyCode(KeyCode.createKeyCode({ first: Key.KEY_A, modifiers: [Modifier.M2] }));
         expect(bindings).to.be.empty;
 
-        bindings = keybindingRegistry.getKeybindingsForKeyCode(KeyCode.createKeyCode({ first: Key.KEY_B, modifiers: [Modifier.M1] }));
+        bindings = keybindingRegistry.getKeybindingsForKeyCode(KeyCode.createKeyCode({ first: Key.KEY_B, modifiers: [Modifier.M2] }));
         expect(bindings).to.be.empty;
 
-        bindings = keybindingRegistry.getKeybindingsForKeyCode(KeyCode.createKeyCode({ first: Key.KEY_C, modifiers: [Modifier.M1] }));
+        bindings = keybindingRegistry.getKeybindingsForKeyCode(KeyCode.createKeyCode({ first: Key.KEY_C, modifiers: [Modifier.M2] }));
         expect(bindings[0].keyCode.key).to.be.equal(validKeyCode.key);
     });
 });
 
 describe("keys api", () => {
-    it("parses a string to a KeyCode correctly", () => {
-        let keycode = KeyCode.parse("ctrl+b");
+    it("should parse a string to a KeyCode correctly", () => {
+        let keycode = KeyCode.parse("shift+b");
         expect(keycode).is.not.undefined;
         if (keycode) {
+            expect(keycode.shift).to.be.true;
             expect(keycode.key).is.equal(Key.KEY_B);
-            expect(keycode.ctrl).to.be.true;
         }
         // Invalid keystroke string
         keycode = KeyCode.parse("ctl+b");
         expect(keycode).is.undefined;
     });
+
+    if (isOSX) {
+        it("should parse a string containing ctrl/option to a KeyCode correctly (OS X)", () => {
+            const keycodeCtrl = KeyCode.parse("ctrl+b");
+            expect(keycodeCtrl).is.not.undefined;
+            if (keycodeCtrl) {
+                expect(keycodeCtrl.meta).to.be.true;
+                expect(keycodeCtrl.key).is.equal(Key.KEY_B);
+            }
+
+            const keycodeOption = KeyCode.parse("option+b");
+            expect(keycodeOption).is.not.undefined;
+            if (keycodeOption) {
+                expect(keycodeOption.alt).to.be.true;
+                expect(keycodeOption.key).is.equal(Key.KEY_B);
+            }
+        });
+    } else {
+        it("should parse a string containing ctrl/option to a KeyCode correctly", () => {
+            const keycode = KeyCode.parse("ctrl+b");
+            expect(keycode).is.not.undefined;
+            if (keycode) {
+                expect(keycode.ctrl).to.be.true;
+                expect(keycode.key).is.equal(Key.KEY_B);
+            }
+
+            const keycodeOption = KeyCode.parse("option+b");
+            expect(keycodeOption).is.not.undefined;
+            if (keycodeOption) {
+                expect(keycodeOption.alt).to.be.true;
+                expect(keycodeOption.key).is.equal(Key.KEY_B);
+            }
+        });
+    }
 });
 
 const TEST_COMMAND: Command = {
@@ -211,9 +247,9 @@ export class TestContribution implements CommandContribution, KeybindingContribu
     }
 
     registerKeybindings(keybindings: KeybindingRegistry): void {
-        const keycode1 = KeyCode.parse('ctrl+a');
-        const keycode2 = KeyCode.parse('ctrl+f1');
-        const keycode3 = KeyCode.parse('ctrl+f2');
+        const keycode1 = KeyCode.parse('shift+a');
+        const keycode2 = KeyCode.parse('shift+f1');
+        const keycode3 = KeyCode.parse('shift+f2');
         if (keycode1 && keycode2 && keycode3) {
             [
                 {

--- a/packages/core/src/common/keybinding.spec.ts
+++ b/packages/core/src/common/keybinding.spec.ts
@@ -1,11 +1,10 @@
-/* tslint:disable:no-unused-expression */
 /*
  * Copyright (C) 2017 Ericsson and others.
  *
  * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
  */
-
+/* tslint:disable:no-unused-expression */
 import { Container, injectable, ContainerModule } from 'inversify';
 import { bindContributionProvider } from './contribution-provider';
 import { ILogger } from './logger';
@@ -13,8 +12,9 @@ import { KeybindingRegistry, KeybindingContext, KeybindingContextRegistry, Keybi
 import { KeyCode, Key, Modifier } from './keys';
 import { CommandRegistry, CommandContribution, Command } from './command';
 import { MockLogger } from './test/mock-logger';
-import { isOSX } from './os';
+import * as os from './os';
 import * as chai from 'chai';
+import * as sinon from 'sinon';
 const jsdom = require('jsdom-global');
 
 const expect = chai.expect;
@@ -24,6 +24,7 @@ chai.config.includeStack = true;
 let keybindingRegistry: KeybindingRegistry;
 let commandRegistry: CommandRegistry;
 let testContainer: Container;
+let stub: sinon.SinonStub;
 
 before(async () => {
     jsdom();
@@ -62,6 +63,11 @@ describe('keybindings', () => {
     beforeEach(() => {
         keybindingRegistry = testContainer.get<KeybindingRegistry>(KeybindingRegistry);
         keybindingRegistry.onStart();
+        stub = sinon.stub(os, 'isOSX').value(false);
+    });
+
+    afterEach(() => {
+        stub.restore();
     });
 
     it("should register the default keybindings", () => {
@@ -75,7 +81,7 @@ describe('keybindings', () => {
     it("should set a keymap", () => {
         const rawKeybindings: RawKeybinding[] = [{
             command: "test.command",
-            keybinding: "shift+c"
+            keybinding: "ctrl+c"
         }];
 
         keybindingRegistry.setKeymap(KeybindingScope.USER, rawKeybindings);
@@ -83,7 +89,7 @@ describe('keybindings', () => {
         const bindings = keybindingRegistry.getKeybindingsForCommand('test.command');
         if (bindings) {
             expect(bindings[0].keyCode.key).to.be.equal(Key.KEY_C);
-            expect(bindings[0].keyCode.shift).to.be.true;
+            expect(bindings[0].keyCode.ctrl).to.be.true;
         }
 
     });
@@ -91,7 +97,7 @@ describe('keybindings', () => {
     it("should reset to default in case of invalid keybinding", () => {
         const rawKeybindings: RawKeybinding[] = [{
             command: "test.command",
-            keybinding: "shift+invalid"
+            keybinding: "ctrl+invalid"
         }];
 
         keybindingRegistry.setKeymap(KeybindingScope.USER, rawKeybindings);
@@ -99,7 +105,7 @@ describe('keybindings', () => {
         const bindings = keybindingRegistry.getKeybindingsForCommand('test.command');
         if (bindings) {
             expect(bindings[0].keyCode.key).to.be.equal(Key.KEY_A);
-            expect(bindings[0].keyCode.shift).to.be.true;
+            expect(bindings[0].keyCode.ctrl).to.be.true;
         }
     });
 
@@ -115,14 +121,14 @@ describe('keybindings', () => {
         if (bindings) {
             expect(bindings.length).to.be.equal(2);
             expect(bindings[0].keyCode.key).to.be.equal(Key.F1);
-            expect(bindings[0].keyCode.shift).to.be.true;
+            expect(bindings[0].keyCode.ctrl).to.be.true;
         }
     });
 
     it("should register a correct keybinding, then default back to the original for a wrong one after", () => {
         let rawKeybindings: RawKeybinding[] = [{
             command: "test.command",
-            keybinding: "shift+c"
+            keybinding: "ctrl+c"
         }];
         // Get default binding
         const keystroke = keybindingRegistry.getKeybindingsForCommand('test.command');
@@ -132,7 +138,7 @@ describe('keybindings', () => {
         const bindings = keybindingRegistry.getKeybindingsForCommand('test.command');
         if (bindings) {
             expect(bindings[0].keyCode.key).to.be.equal(Key.KEY_C);
-            expect(bindings[0].keyCode.shift).to.be.true;
+            expect(bindings[0].keyCode.ctrl).to.be.true;
         }
 
         // Set invalid binding
@@ -154,37 +160,37 @@ describe('keybindings', () => {
     it("should only return the more specific keybindings when a keystroke is entered", () => {
         const rawKeybindingsUser: RawKeybinding[] = [{
             command: "test.command",
-            keybinding: "shift+b"
+            keybinding: "ctrl+b"
         }];
 
         keybindingRegistry.setKeymap(KeybindingScope.USER, rawKeybindingsUser);
 
         const rawKeybindingsSpecific: RawKeybinding[] = [{
             command: "test.command",
-            keybinding: "shift+c"
+            keybinding: "ctrl+c"
         }];
 
-        const validKeyCode = KeyCode.createKeyCode({ first: Key.KEY_C, modifiers: [Modifier.M2] });
+        const validKeyCode = KeyCode.createKeyCode({ first: Key.KEY_C, modifiers: [Modifier.M1] });
 
         keybindingRegistry.setKeymap(KeybindingScope.WORKSPACE, rawKeybindingsSpecific);
 
-        let bindings = keybindingRegistry.getKeybindingsForKeyCode(KeyCode.createKeyCode({ first: Key.KEY_A, modifiers: [Modifier.M2] }));
+        let bindings = keybindingRegistry.getKeybindingsForKeyCode(KeyCode.createKeyCode({ first: Key.KEY_A, modifiers: [Modifier.M1] }));
         expect(bindings).to.be.empty;
 
-        bindings = keybindingRegistry.getKeybindingsForKeyCode(KeyCode.createKeyCode({ first: Key.KEY_B, modifiers: [Modifier.M2] }));
+        bindings = keybindingRegistry.getKeybindingsForKeyCode(KeyCode.createKeyCode({ first: Key.KEY_B, modifiers: [Modifier.M1] }));
         expect(bindings).to.be.empty;
 
-        bindings = keybindingRegistry.getKeybindingsForKeyCode(KeyCode.createKeyCode({ first: Key.KEY_C, modifiers: [Modifier.M2] }));
+        bindings = keybindingRegistry.getKeybindingsForKeyCode(KeyCode.createKeyCode({ first: Key.KEY_C, modifiers: [Modifier.M1] }));
         expect(bindings[0].keyCode.key).to.be.equal(validKeyCode.key);
     });
 });
 
 describe("keys api", () => {
     it("should parse a string to a KeyCode correctly", () => {
-        let keycode = KeyCode.parse("shift+b");
+        let keycode = KeyCode.parse("ctrl+b");
         expect(keycode).is.not.undefined;
         if (keycode) {
-            expect(keycode.shift).to.be.true;
+            expect(keycode.ctrl).to.be.true;
             expect(keycode.key).is.equal(Key.KEY_B);
         }
         // Invalid keystroke string
@@ -192,39 +198,69 @@ describe("keys api", () => {
         expect(keycode).is.undefined;
     });
 
-    if (isOSX) {
-        it("should parse a string containing ctrl/option to a KeyCode correctly (OS X)", () => {
-            const keycodeCtrl = KeyCode.parse("ctrl+b");
-            expect(keycodeCtrl).is.not.undefined;
-            if (keycodeCtrl) {
-                expect(keycodeCtrl.meta).to.be.true;
-                expect(keycodeCtrl.key).is.equal(Key.KEY_B);
-            }
+    it("should parse a string containing special modifiers to a KeyCode correctly", () => {
+        const stub = sinon.stub(os, 'isOSX').value(false);
 
-            const keycodeOption = KeyCode.parse("option+b");
-            expect(keycodeOption).is.not.undefined;
-            if (keycodeOption) {
-                expect(keycodeOption.alt).to.be.true;
-                expect(keycodeOption.key).is.equal(Key.KEY_B);
-            }
-        });
-    } else {
-        it("should parse a string containing ctrl/option to a KeyCode correctly", () => {
-            const keycode = KeyCode.parse("ctrl+b");
-            expect(keycode).is.not.undefined;
-            if (keycode) {
-                expect(keycode.ctrl).to.be.true;
-                expect(keycode.key).is.equal(Key.KEY_B);
-            }
+        const keycode = KeyCode.parse("ctrl+b");
+        expect(keycode).is.not.undefined;
+        if (keycode) {
+            expect(keycode.ctrl).to.be.true;
+            expect(keycode.key).is.equal(Key.KEY_B);
+        }
 
-            const keycodeOption = KeyCode.parse("option+b");
-            expect(keycodeOption).is.not.undefined;
-            if (keycodeOption) {
-                expect(keycodeOption.alt).to.be.true;
-                expect(keycodeOption.key).is.equal(Key.KEY_B);
-            }
-        });
-    }
+        const keycodeOption = KeyCode.parse("option+b");
+        expect(keycodeOption).is.not.undefined;
+        if (keycodeOption) {
+            expect(keycodeOption.alt).to.be.true;
+            expect(keycodeOption.key).is.equal(Key.KEY_B);
+        }
+
+        const keycodeCommand = KeyCode.parse("cmd+b");
+        expect(keycodeCommand).is.undefined;
+
+        const keycodeCtrlOrCommand = KeyCode.parse("ControlOrCommand+b");
+        expect(keycodeCtrlOrCommand).is.not.undefined;
+        if (keycodeCtrlOrCommand) {
+            expect(keycodeCtrlOrCommand.meta).to.be.false;
+            expect(keycodeCtrlOrCommand.ctrl).to.be.true;
+            expect(keycodeCtrlOrCommand.key).is.equal(Key.KEY_B);
+        }
+
+        stub.restore();
+    });
+
+    it("should parse a string containing special modifiers to a KeyCode correctly (macOS)", () => {
+        stub = sinon.stub(os, 'isOSX').value(true);
+        const keycode = KeyCode.parse("ctrl+b");
+        expect(keycode).is.not.undefined;
+        if (keycode) {
+            expect(keycode.ctrl).to.be.true;
+            expect(keycode.key).is.equal(Key.KEY_B);
+        }
+
+        const keycodeOption = KeyCode.parse("option+b");
+        expect(keycodeOption).is.not.undefined;
+        if (keycodeOption) {
+            expect(keycodeOption.alt).to.be.true;
+            expect(keycodeOption.key).is.equal(Key.KEY_B);
+        }
+
+        const keycodeCommand = KeyCode.parse("cmd+b");
+        expect(keycodeCommand).is.not.undefined;
+        if (keycodeCommand) {
+            expect(keycodeCommand.meta).to.be.true;
+            expect(keycodeCommand.key).is.equal(Key.KEY_B);
+        }
+
+        const keycodeCtrlOrCommand = KeyCode.parse("ControlOrCommand+b");
+        expect(keycodeCtrlOrCommand).is.not.undefined;
+        if (keycodeCtrlOrCommand) {
+            expect(keycodeCtrlOrCommand.meta).to.be.true;
+            expect(keycodeCtrlOrCommand.ctrl).to.be.false;
+            expect(keycodeCtrlOrCommand.key).is.equal(Key.KEY_B);
+        }
+        stub.restore();
+    });
 });
 
 const TEST_COMMAND: Command = {
@@ -247,9 +283,9 @@ export class TestContribution implements CommandContribution, KeybindingContribu
     }
 
     registerKeybindings(keybindings: KeybindingRegistry): void {
-        const keycode1 = KeyCode.parse('shift+a');
-        const keycode2 = KeyCode.parse('shift+f1');
-        const keycode3 = KeyCode.parse('shift+f2');
+        const keycode1 = KeyCode.parse('ctrl+a');
+        const keycode2 = KeyCode.parse('ctrl+f1');
+        const keycode3 = KeyCode.parse('ctrl+f2');
         if (keycode1 && keycode2 && keycode3) {
             [
                 {

--- a/packages/core/src/common/keys.ts
+++ b/packages/core/src/common/keys.ts
@@ -72,13 +72,25 @@ export class KeyCode {
             }
             const key = EASY_TO_KEY[keyString];
 
-            if (keyString === 'meta') {
+            /* meta only works on macOS */
+            if (keyString === SpecialCases.META) {
+                if (isOSX) {
+                    sequence.push(`${Modifier.M1}`);
+                } else {
+                    return undefined;
+                }
+                /* ControlOrCommand for keybindings that work on both macOS and other platforms */
+            } else if (keyString === SpecialCases.CTRL_OR_COMMAND) {
                 sequence.push(`${Modifier.M1}`);
-            }
-            if (Key.isKey(key)) {
+            } else if (Key.isKey(key)) {
                 if (Key.isModifier(key.code)) {
                     if (key.keyCode === EasyKey.CONTROL.keyCode) {
-                        sequence.push(`${Modifier.M1}`);
+                        // CTRL on MacOS X (M4)
+                        if (isOSX) {
+                            sequence.push(`${Modifier.M4}`);
+                        } else {
+                            sequence.push(`${Modifier.M1}`);
+                        }
                     } else if (key.keyCode === EasyKey.SHIFT.keyCode) {
                         sequence.push(`${Modifier.M2}`);
                     } else if (key.keyCode === EasyKey.ALT.keyCode) {
@@ -201,8 +213,15 @@ const SPECIAL_ALIASES: { [index: string]: string } = {
     'esc': 'escape',
     'mod': 'ctrl',
     'ins': 'insert',
-    'del': 'delete'
+    'del': 'delete',
+    'control': 'ctrl',
+    'CommandOrControl': 'ControlOrCommand',
 };
+
+export namespace SpecialCases {
+    export const META = 'meta';
+    export const CTRL_OR_COMMAND = 'ControlOrCommand';
+}
 
 export namespace EasyKey {
     export const ENTER: EasyKey = { keyCode: 13, easyString: 'enter' };

--- a/packages/keymaps/README.md
+++ b/packages/keymaps/README.md
@@ -20,7 +20,7 @@ Example of a valid `keymaps.json` file
 
 For most keys you can directly use the name of the key i.e `a`, `3`,  `/`, `-`.
 
-You can use `shift`, `ctrl`, `alt`, `meta`, `option` (`alt`), `command` (`meta`), `cmd` (`meta`) as modifiers.
+You can use `shift`, `ctrl`, `alt`, `meta`, `option` (`alt`), `command` (`meta`), `cmd` (`meta`) as modifiers. You can also `ControlOrCommand` or `CommandOrControl` as modifiers if you want the shortcut to fallback on `ctrl` coming from macOS. Note that if you defined a custom shortcut with `cmd`, `command` or `meta`, the same keymaps file won't work on a Windows/Linux machine as this key doesn't have an equivalent.
 
 You can also use the following strings for special keys: `backspace`, `tab`, `enter`, `return`, `capslock`, `esc`, `escape`, `space`, `pageup`, `pagedown`, `end`, `home`, `left`, `up`, `right`, `down`, `ins`, `del` and `plus`.
 

